### PR TITLE
contrib: Adjustments to AppArmor docker engine policy for docker 1.9

### PR DESCRIPTION
**Pull request by stefanberger**
This is a set of docker engine policy adjustments needed for docker 1.9.

### CODE REVIEW

**Comment by stefanberger** [comment]
@ewindisch Please let me know of comments.

**Comment by stefanberger** [comment]
Since the docker version from yesterday (10/7), I get the following DENIED messages in the audit.log for which I haven't been able to find an AppArmor rule yet:

type=AVC msg=audit(1444312406.311:58497): apparmor="DENIED" operation="mount" info="failed srcname match" error=-13 profile="/usr/bin/docker" name="/.pivot_root501395159/" pid=1156421 comm="exe" flags="rw, rprivate"

**Comment by thaJeztah** [comment]
ping @ewindisch for review, (also let me know if this should be added to the milestone)

File: contrib/apparmor/docker-engine [hunk 1]
```diff
@@ -172,7 +182,7 @@ profile /usr/bin/docker (attach_disconnected, complain) {
   profile /sbin/tune2fs (complain) {
     /sbin/tune2fs rm,
 
-    /lib/** r,
+    /lib/** rm,
```
Comment (ewindisch) [inline contrib/apparmor/docker-engine]: any idea why we suddenly need 'm' on /lib everywhere?

**Comment by ewindisch** [comment]
@thaJeztah it'd be nice to have this working in 1.9, yes.

@stefanberger just a question about the 'm' on /lib/** but okay otherwise. Your remaining error might be solved with either 'mount' or 'pivot_root'  statements (these can be restricted to paths)

**Comment by stefanberger** [comment]
The m on the /lib/** come from using golang 1.4.2. I had not needed it before, either.

For that one error, which occurs during a 'docker build',I have now all of the following rules in a policy; none of them seems to be the rule addressing the underlying requirement.

  mount options=rw,rprivate /.pivot_root[0-9]*,      
  mount options=rw,rprivate /.pivot_root[0-9]*/*,    
  mount options=rw,rprivate /.pivot_root[0-9]*/**,   
  mount options=rw,rprivate -> /.pivot_root[0-9]*,   
  mount options=rw,rprivate -> /.pivot_root[0-9]*/*, 
  mount options=rw,rprivate -> /.pivot_root[0-9]*/**,
  mount options=rw,rprivate /.pivot_root*,      
  mount options=rw,rprivate /.pivot_root*/*,    
  mount options=rw,rprivate /.pivot_root*/**,   
  mount options=rw,rprivate -> /.pivot_root*,   
  mount options=rw,rprivate -> /.pivot_root*/*, 
  mount options=rw,rprivate -> /.pivot_root*/**,
  /.pivot_root*/ rw,  
  /.pivot_root**/ rw, 
  /.pivot_root*/* rw, 
  /.pivot_root*/** rw,
  pivot_root /.pivot_root[0-9]*/*,  
  pivot_root /.pivot_root[0-9]*/**, 
  pivot_root /.pivot_root[0-9]**/**,
  pivot_root oldroot=/.pivot_root[0-9]*/*,  
  pivot_root oldroot=/.pivot_root[0-9]*/**, 
  pivot_root oldroot=/.pivot_root[0-9]**/**,

**Comment by stefanberger** [comment]
The star and star star in the rules in the message above were unfortunately mangled..

**Comment by jfrazelle** [comment]
LGTM

File: contrib/apparmor/docker-engine (hunk 1 above)
Comment (jfrazelle) [inline contrib/apparmor/docker-engine]: are those memory executables? idk

**Comment by jfrazelle** [comment]
ping @ewindisch

**Comment by ewindisch** [comment]
LGTM.  But lets follow up with a patch for mount errors

**Comment by jfrazelle** [comment]
btw the mount comes from https://github.com/opencontainers/runc/blob/0f137226d88eb79803f69300e7f46ce24b103d02/libcontainer/rootfs_linux.go#L536 For reference

**Comment by stefanberger** [comment]
@ewindisch: I spent a long time trying to find the right rule... to no avail. Maybe someone else can give it a try.

**Pull request merged**